pricegraph/src/orderbook.rs

```diff
@@ -347,14 +347,18 @@ impl Orderbook {
     ///
     /// Note that the limit price is expressed as an exchange limit price, i.e.
     /// with implicitely included fees.
-    pub fn fill_order_at_price(&mut self, pair: TokenPair, actual_limit_price: f64) -> f64 {
+    pub fn fill_order_at_price(&mut self, pair: TokenPair, limit_price: f64) -> f64 {
         if !self.is_token_pair_valid(pair) {
             return 0.0;
         }
         self.update_projection_graph();
 
         let (sell, buy) = (node_index(pair.sell), node_index(pair.buy));
-        let effective_limit_price = actual_limit_price / FEE_FACTOR;
+        // NOTE: The limit price is expressed in `buy_amount / sell_amount` for
+        // the specified token pair, but since we are finding transitive orders
+        // in the opposite direction to match this order, calculate the maximum
+        // price that still respects the provided limit price.
+        let max_price = 1.0 / (limit_price * FEE_FACTOR);
```

Comment:
If I understand correctly, we multiply with the fee factor (1.001) to tighten the limit price and thus allow for the fee of the counter order. Is this right? Maybe it's worth documenting this. I first read this as we are deducting the fee (but we are actually doubling it).

Reply:
This is because there are two mesurements for prices, exchange prices (that have implicit fees) and graph prices (called effective prices in the code) that have explicit fees.

By this I mean, to get the buy amount amount for an exchange price:
```
buy_amount = sell_amount * (exchange_price / FEE_FACTOR)
```

While to get the buy amount from an effective price:
```
buy_amount = sell_amount * effective_price
```

Whats confusing is how I wrote this equation. We want to convert the inverted price to an effective price, so it would probably be more clear if written:
```
(1.0 / limit_price) / FEE_FACTOR;
```

> Maybe it's worth documenting this.

This should already be in the doc comments:
```rust
    /// Note that the limit price is expressed as an exchange limit price, i.e.
    /// with implicitely included fees.
```

Maybe I can elaborate a bit more.

> I first read this as we are deducting the fee (but we are actually doubling it).

I don't think this doubles the fees, just adds it once. The subtlety is that we are converting the inverted price from an exchange price to a effective price.